packages/client/src/sigstore.ts

```diff
@@ -81,6 +81,31 @@ export async function verify(
   return verifier.verify(deserializedBundle, opts, payload);
 }
 
+export interface BundleVerifier {
+  verify(bundle: sigstore.SerializedBundle): void;
+}
+
+export async function createVerifier(
+  options: config.BasicVerifierOptions
+): Promise<BundleVerifier> {
+  const trustedRoot = await tuf.getTrustedRoot({
```

Comment:
What are the failure cases for the tuf trust root becoming stale/out of date? Does calling verify trigger a refresh if needed or does it fail?

Reply:
The worst that could happen here would be if someone were using the verifier at the same time that Sigstore's "trusted_root.json" file were updated. If that update included the revocation or expiration of a previously valid key we might incorrectly verify a bundle that should be considered invalid.

Unless someone were to instantiate a verifier and use it in some extremely long-running process, I can't see this being a problem. In the npm CLI use case, the TUF data will be refreshed with each no invocation of `npm audit signatures`